src/components/Common/Toast.tsx

```diff
@@ -0,0 +1,60 @@
+import { Alert, Snackbar } from "@mui/material";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import {
+  clearCurrentToast,
+  dequeueToast,
+  setToastOpen,
+} from "../../state/slices/toastSlice";
+import { useEffect } from "react";
+
+const Toast = () => {
+  const { currentToast, loaf, open } = useAppSelector((state) => state.toast);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (loaf.length > 0 && !currentToast) {
+      // Show next the toast if one isn't active
+      dispatch(dequeueToast());
+      dispatch(setToastOpen(true));
+    } else if (loaf.length > 0 && currentToast && open) {
+      // Close the current toast when a new one is added
+      dispatch(setToastOpen(false));
+    }
+  }, [currentToast, loaf, dispatch]);
+
+  const handleClose = (
+    event?: React.SyntheticEvent | Event,
```

Comment:
I was gonna ask since it isnt being used, what was the vision for it? I am trying to understand this whole toast thing

Reply:
@christopherkinyua this is an event argument that gets supplied from MUI's callback, so we don't have control over the function signature. That event argument is going to be there whether we use it or not, but since we need access to the second `reason` argument they give us we need to include it in the function arguments

@NasihNazeem I remember I was having some typescript issues before which is why I kept this here, but for some reason it's not complaining anymore lol so I've changed it to `_`